frontend/public/module/k8s/k8s-actions.js

```diff
@@ -155,34 +153,43 @@ const actions = {
     dispatch({type: types.watchK8sList, id, query});
     REF_COUNTS[id] = 1;
 
-    // Fetch entire list (XHR) then use its resourceVersion to
-    //  start listening on a WS (?resourceVersion=$resourceVersion)
-    //  start the process over when:
-    //   1. the WS closes abnormally
-    //   2. the WS can not establish a connection within $TIMEOUT
+    /** @type {(continueToken: string) => Promise<string>} */
+    const incrementallyLoad = async(continueToken = '') => {
+      const response = await k8sList(k8skind, {...query, limit: paginationLimit, ...(continueToken ? {continue: continueToken} : {})}, true);
+
+      if (response.metadata.continue) {
+        dispatch(actions.bulkAddToList(id, response.items));
+        return incrementallyLoad(response.metadata.continue);
+      }
+      dispatch(actions.loaded(id, response.items));
```

Comment:
Don't we want to dispatch loaded with all the items, not just the ones in the last response? In my experimentation, this causes the list to only show the last bit of data it fetched. eg: set paginationLimit to 10 and try to load all pods. Here's a video of it happening: https://i.imgur.com/NQKiEqL.gifv

Reply:
Good catch!, but we'd need to change the behavior of the reducer to do that.  Always dispatching `loaded` first would probably be easier and we'd incrementally render stuff, too.